core/sql/executor/ex_hash_grby.cpp

```diff
@@ -452,6 +463,8 @@ short ex_hash_grby_tcb::work() {
             hasFreeTupp_ = TRUE;
     	};
       };
+      if (bmoStats_)
+         bmoStats_->setBmoPhase(PHASE_END-HGB_READ_PHASE);
```

Comment:
For your latter question, I didn't want to disturb the existing enum values and hence kept the existing value. But, for BMO stats, I wanted show the phase of the instance that hasn't moved to the next higher phase.  When the BMO stats is merged from different esp instances,  higher phase value is used. Hence, the phase enum value is reversed for BMO stats